dspace-api/src/test/java/org/dspace/content/RelationshipMetadataServiceTest.java

```diff
@@ -0,0 +1,140 @@
+     *
+     * Other methods can be annotated with @After here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @After
+    @Override
+    public void destroy() {
+        context.abort();
+        super.destroy();
+    }
+
+
+    @Test
+    public void testGetRelationshipMetadata() throws Exception {
+        context.turnOffAuthorisationSystem();
+        itemService.addMetadata(context, item, "relationship", "type", null, null, "Publication");
+        itemService.addMetadata(context, authorItem, "relationship", "type", null, null, "Author");
+        itemService.addMetadata(context, authorItem, "person", "familyName", null, null, "familyName");
+        itemService.addMetadata(context, authorItem, "person", "givenName", null, null, "firstName");
+        EntityType publicationEntityType = entityTypeService.create(context, "Publication");
+        EntityType authorEntityType = entityTypeService.create(context, "Author");
+        RelationshipType isAuthorOfPublication = relationshipTypeService
+            .create(context, publicationEntityType, authorEntityType, "isAuthorOfPublication", "isPublicationOfAuthor",
+                    null, null, null, null);
+
+        Relationship relationship = relationshipService.create(context, item, authorItem, isAuthorOfPublication, 0, 0);
+        List<MetadataValue> authorList = itemService.getMetadata(item, "dc", "contributor", "author", Item.ANY);
+        assertThat(authorList.size(), equalTo(1));
+        assertThat(authorList.get(0).getValue(), equalTo("familyName, firstName"));
+
+        List<MetadataValue> relationshipMetadataList = itemService
+            .getMetadata(item, "relation", "isAuthorOfPublication", null, Item.ANY);
+        assertThat(relationshipMetadataList.size(), equalTo(1));
+        assertThat(relationshipMetadataList.get(0).getValue(), equalTo(String.valueOf(authorItem.getID())));
+
+        List<RelationshipMetadataValue> list = relationshipMetadataService.getRelationshipMetadata(item, true);
+        assertThat(list.size(), equalTo(2));
+        assertThat(list.get(0).getValue(), equalTo("familyName, firstName"));
+        assertThat(list.get(0).getMetadataField().getMetadataSchema().getName(), equalTo("dc"));
+        assertThat(list.get(0).getMetadataField().getElement(), equalTo("contributor"));
+        assertThat(list.get(0).getMetadataField().getQualifier(), equalTo("author"));
+        assertThat(list.get(0).getAuthority(), equalTo("virtual::" + relationship.getID()));
+
+        assertThat(list.get(1).getValue(), equalTo(String.valueOf(authorItem.getID())));
+        assertThat(list.get(1).getMetadataField().getMetadataSchema().getName(), equalTo("relation"));
+        assertThat(list.get(1).getMetadataField().getElement(), equalTo("isAuthorOfPublication"));
+        assertThat(list.get(1).getAuthority(), equalTo("virtual::" + relationship.getID()));
+
+        context.restoreAuthSystemState();
```

Comment:
This restoration of authorization checks should be moved *before* the tests of `getRelationshipMetadata()`....otherwise, we are ignoring authorization for the entirety of the tests.  As a best practice, authorization should only be turned off temporarily to create test content -- it should never remain off for an entire test. So, just for safety, please move this to around line 116 (i.e. just after you are done creating your test data, but *before* any `assert` checks are performed).

Reply:
The `context.restoreAuthSystemState()` has been moved up